### PR TITLE
feat: 🎸 customize actions on reference editors

### PR DESCRIPTION
**Pull request by anho**
Add render prop to provide override for reference actions.

### CODE REVIEW

**Review by DanweDE — APPROVED** [review]

File: packages/reference/src/common/ReferenceEditor.tsx [hunk 1]
```diff
@@ -16,6 +17,7 @@ export interface ReferenceEditorProps {
   sdk: FieldExtensionSDK;
   viewType: ViewType;
   renderCustomCard?: (props: CustomEntryCardProps) => React.ReactElement | false;
+  renderCustomActions?: (props: CustomActionProps) => React.ReactElement;
```
Comment (suevalov) [inline packages/reference/src/common/ReferenceEditor.tsx:20]: Maybe it's worth to support the same `false` logic for consistency with `renderCustomCard` property?

**Review by suevalov — APPROVED** [review]

File: packages/reference/src/common/ReferenceEditor.tsx (hunk 1 above)
Comment (anho) [reply packages/reference/src/common/ReferenceEditor.tsx:20]: That is what I had in the beginning. After asking @DanweDE about the rationale behind the support for false for the card I didn't feel like we actually need it.

**Review by giotiskl — APPROVED** [review]

File: packages/reference/src/components/LinkActions/LinkActions.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as styles from './styles';
 import { EntityType, ContentType, ActionLabels } from '../../types';
 import { CreateEntryLinkButton } from '../CreateEntryLinkButton/CreateEntryLinkButton';
 
-interface LinkActionsProps {
+export interface LinkActionsProps {
```
Comment (DanweDE) [inline packages/reference/src/components/LinkActions/LinkActions.tsx:7]: I think we should actually export this in the main `index.tsx` just like we expose the `CustomEntryCardProps` [here](https://github.com/contentful/field-editors/blob/fd1d9353d459f96f255430eba0f6b5217c30f973/packages/reference/src/index.tsx#L16)

As we're making this a public interface, we should also clean it up a bit. E.g. `multiple` is not consistent with most our public naming, e.g. `isMultipleLinks: boolean` would be a bit clearer.

A nice property to add (not sure if you need it though right now) would be one to indicate the current number of links to e.g. render different action buttons for an empty state link field editor.

Comment (anho) [reply packages/reference/src/components/LinkActions/LinkActions.tsx:7]: All of the things you mentioned are good points, I'd skip them for time being and once we see actual need for these "features" I'd happily come back and implement them.

Comment (DanweDE) [reply packages/reference/src/components/LinkActions/LinkActions.tsx:7]: The first two points are not features, I think you need to export `LinkActionsProps` from the main package to be able to use it. Renaming `multiple` should be done for consistency/clarity.

**Pull request merged**